File: Hackerrank/sock-merchant.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the sockMerchant function below.
def sockMerchant(n, ar):
    distinctElements=set(ar)
    pairs=0
    for ele in distinctElements:
        cnt=ar.count(ele)
        pairs+=int(cnt/2)
    return pairs
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    ar = list(map(int, input().rstrip().split()))

    result = sockMerchant(n, ar)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Hackerrank/sock-merchant.js
function sockMerchant(n, ar) {
  let socks = {};
  let pairs = 0;
for (let element of ar) {
  socks[element] = socks[element] + 1 || 1;
    if (socks[element] % 2 === 0) {
      pairs += 1;
    }
  }
return pairs;
}